ocaml/xapi/quicktest_cbt.ml

```diff
@@ -0,0 +1,153 @@
+(*
+ * Copyright (C) 2017 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Client
+open Quicktest_common
+exception Test_assertion_failed of string
+
+let start session_id =
```

Comment:
A general comment. By making most functions local to `start`, this is more difficult to test. Since this a module of its own, I don't see a need for making functions local.

Reply:
To be honest, localising to start is only useful for acquiring `session_id`. I'm trying to do away with this, but it's a bit confusing, as setting `let session_id = Quicktest.s` creates a cyclic dependency jbuilder doesn't like. The only thing I can think of is adding `session_id` to every other function, but that seems less clean to me. What do you think?

Reply:
I would pass around `session_id`. It's not nice but I would pay this price to untangle the functions.